src/runway/sources/source.py

```diff
@@ -0,0 +1,17 @@
+"""
+    Abstract parent class for a 'Source' type object.
+    Allows us to specify specific remote sourced resources for out application
+    (Git, S3, ect.)
+"""
+class Source(object):
```

Comment:
I would add a `fetch()` (or whatever we want to name it) method here to be implemented in subclasses as the single method used across all of them for getting.

```python
    def fetch(self):
        """Retrieve remote source.

        To be implemented in each subclass.

        """
        raise NotImplementedError
```

Reply:
This is very similar to how I used to do it in Ruby, good call.